include/seqan3/alphabet/gap/gap.hpp

```diff
@@ -0,0 +1,146 @@
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+// Author: Hannes Hauswedell <hannes.hauswedell@fu-berlin.de>
+// Author: Marcel Ehrhardt <marcel.ehrhardt@fu-berlin.de>
+// Author: David Heller <david.heller@fu-berlin.de>
+// ==========================================================================
+
+#pragma once
+
+#include <optional>
+
+#include "../alphabet.hpp"
+
+/*! The gap alphabet
+ * \ingroup alphabet
+ */
+
+namespace seqan3
+{
+
+/*! The gap alphabet of -
+ *
+ * The alphabet always has the same value ('-').
+ *
+ *     gap my_gap{};
+ *     gap another_gap{}.from_char('A'); // setting this does not change anything
+ *
+ *     if (my_gap.to_char() == another_gap.to_char())
+ *        std::cout << "Both gaps are the same!";
+ */
+
+struct gap
+{
+    /* types */
+    //! the type of the alphabet when converted to char (e.g. via @link to_char @endlink)
+    using char_type = char;
+    //! the type of the alphabet when represented as a number (e.g. via @link to_integral @endlink)
+    using integral_type = bool;
+    using c_type = bool;
+
+    /* member */
+    //! internal value
+    static constexpr c_type value = 0;
+
+    //! The size of the alphabet, i.e. the number of different values it can take.
+    static constexpr integral_type value_size{1};
+
+    /* public member functions */
+    //! ability to cast to @link char_type @endlink **explicitly**.
+    explicit constexpr operator char_type() const
+    {
+        return to_char();
+    }
+
+    //! return the letter as a character of @link char_type @endlink.
+    constexpr char_type to_char() const
+    {
+        return '-';
+    }
+
+    //! return the gap's numeric value.
+    constexpr integral_type to_integral() const
+    {
+        return static_cast<integral_type>(value);
+    }
+
+    //! assign from a character does not do anything.
+    constexpr gap & from_char(char_type const in)
+    {
+        return *this;
+    }
+
+    //! assign from a numeric value does not do anything.
+    constexpr gap & from_integral(integral_type const in)
+    {
+        return *this;
+    }
+
+    //! @name comparison operators
+    //!@{
+    constexpr bool operator==(gap const & rhs) const
+    {
+        return value == rhs.value;
+    }
+
+    constexpr bool operator!=(gap const & rhs) const
+    {
+        return value != rhs.value;
+    }
+
+    constexpr bool operator<(gap const & rhs) const
+    {
+        return value < rhs.value;
+    }
+
+    constexpr bool operator>(gap const & rhs) const
+    {
+        return value > rhs.value;
+    }
+
+    constexpr bool operator<=(gap const & rhs) const
+    {
+        return value <= rhs.value;
+    }
+
+    constexpr bool operator>=(gap const & rhs) const
+    {
+        return value >= rhs.value;
+    }
+    //!@}
+};
+
```

Comment:
wrap the static_assert in an `#ifndef NDEBUG` and also test the inner_alphabet type. See dna4 for more details